impl/README.md

```diff
@@ -0,0 +1,188 @@
+
+```text
+implementation("io.serverlessworkflow:serverlessworkflow-impl-core:7.0.0-SNAPSHOT")
+```
+
+And only if your workflow is using HTTP calls, you must add:
+
+```text
+implementation("io.serverlessworkflow:serverlessworkflow-impl-http:7.0.0-SNAPSHOT")
+```
+
+## How to use
+
+The quick version is intended for impatient users who want to try something as soon as possible.
+
+The detailed version is more suitable for those users interested in a more thoughtful discussion of the API.
+
+### Quick version
+
+For a quick introduction, we will use a simple workflow [definition](../examples/simpleGet/src/main/resources/get.yaml) that performs a get call. 
+We are going to show two ways of invoking the workflow: 
+  - blocking the thread till the get request goes through
+  - returning control to the caller, so the main thread continues while the get is executed
+
+In order to execute the workflow, blocking the thread till the HTTP request is completed, you should write
+
+``` java 
+try (WorkflowApplication appl = WorkflowApplication.builder().build()) {
+      logger.info(
+          "Workflow output is {}",
+          appl.workflowDefinition(WorkflowReader.readWorkflowFromClasspath("get.yaml"))
+              .instance(Map.of("petId", 10))
+              .start()
+              .join());
+    }
+```
+You can find the complete java code [here](../examples/simpleGet/src/main/java/BlockingExample.java)
+
+In order to execute the workflow without blocking the calling thread till the HTTP request is completed, you should write
+
+``` java 
+  try (WorkflowApplication appl = WorkflowApplication.builder().build()) {
+      appl.workflowDefinition(WorkflowReader.readWorkflowFromClasspath("get.yaml"))
+          .instance(Map.of("petId", 10))
+          .start()
+          .thenAccept(node -> logger.info("Workflow output is {}", node));
+    }
+```
+When the HTTP request is done, both examples will print a similar output
+
+
+```shell
+Workflow output is {"id":10,"category":{"id":10,"name":"string"},"name":"doggie","photoUrls":["string"],"tags":[{"id":10,"name":"string"}],"status":"string"}
+
+You can find the complete java code [here](../examples/simpleGet/src/main/java/NotBlockingExample.java)
+
+### Detailed version
+
+To discuss runtime API we are going to use a couple of workflow:
+- [listen.yaml](../examples/events/src/main/listen.yaml), which waits for an event reporting a temperature greater than 38
+- [emit.yaml](../examples/events/src/main/emit.yaml), which emits events with a certain temperature, specified as workflow parameter.
+
+Here is a summary of what we are trying to do: We will start the `listen.yaml` workflow, which will be completed when it receives an event with the proper temperature. It won't block the main thread while waiting for it. Then, we will send an event with a lower temperature that will be ignored. Finally, we will send an event with a higher temperature to complete the waiting workflow. 
```

Comment:
You can break down in several bullets:
- The listen.yaml workflow waits for an event (not-blocking).
- We send an event with a low temperature (ignored).
- We send an event with a high temperature (completes the workflow).